app/gameengine.py

```diff
@@ -5461,9 +5604,16 @@ def choose_cards_cleanup_remaining(self, performing_player_id, remaining_card_id
                         "card_ids": remaining_card_ids,
                         "from_zone": from_zone,
                         "to_zone": to_zone,
-                        "bottom": True,
+                        "bottom": bottom,
                         "continuation": continuation,
                     })
+                case "remove_choice_from_last_revealed_cards":
+                    last_revealed_cards = []
```

Comment:
This feels like you're doing it backwards.
Don't you just want to iterate through player.last_revealed_cards and if you find the matching id, then you .erase it and break?
Maybe I'm missing something.

Reply:
At this point, we're only working with the remaining card ids; the choice was already removed and the list is being recreated from the remaining card ids.

Reply:
Okay, I see where I was mistaken, but I guess I still don't get it.
Does player.last_revealed_cards not already have the cards in it at this point?
Presumably you revealed all the cards so they're in there, then the user got a choose cards and picked one.
Now remaining_card_ids should be all except that one.
So my first thought would be to iterate through player.last_revealed_cards and remove any that are not in remaining_card_ids?

Functionally the same as rebuilding the list like you're doing, but semantically better matches the name of the "remove_choice...*" string.

Thinking about future me reading this later, we probably at least want a comment here explaining what we're doing.
"# Filter out chosen cards by rebuilding the last_revealed_cards list to only include remaining cards." maybe?

Reply:
Yeah. The wording matches better with how the definition is read rather than what it actually does under the hood.
- You have a choice then your remaining action is removing that choice from the revealed cards.

I'll add a comment for clarity.